### ISSUE TITLE
Race condition on generated filenames using timestamps

### ISSUE THREAD

**Issue by faustinoaq** [amberframework/micrate#31]
Hi @amberframework/contributors This issue is causing Travis Failures

Please, see https://github.com/amberframework/amber/pull/775#issuecomment-385537503

I suggest to fix it by using incremental names or wait at least `1.millisecond` before generate migration file.

**Comment by faustinoaq** [comment]
The incremental names approach was suggested by @vladfaust on https://github.com/amberframework/micrate/issues/20

**Comment by faustinoaq** [comment]
Another approach to solve this is by waiting at least `1.millisecond` here before generating a migration file, like:

```cr
 sleep 1.millisecond
 timestamp = time.to_s("%Y%m%d%H%M%S") 
 filename = File.join(migrations_path, "#{timestamp}_#{name}.sql") 
```

on:

https://github.com/amberframework/micrate/blob/4168d23e8e8d4aaad4852b08221e63628a891ae5/src/micrate.cr#L65-L66

WDYT?

**Comment by robacarp** [comment]
Sleeping before creating the file isn't going to prevent a race condition, it's just going to move it down the road a little bit. Generating a file with a timestamp, no matter how specific that timestamp, is always going to be subject to a race condition. Adding more digits to the migration timestamp isn't the answer either.

What about adding a "does this file exist?" check in the [Micrate.create](https://github.com/amberframework/micrate/blob/master/src/micrate.cr#L64-L81) method? That will still be subject to a race condition when multiple migrations are generated concurrently, but running two commands consecutively will be safe.

**Comment by drujensen** [comment]
What about using a UUID instead of a timestamp?

**Comment by faustinoaq** [comment]
Interesting idea, Thank you for sharing it, I going to try that as well :sweat_smile:

**Comment by faustinoaq** [comment]
Already fixed on micrate `v0.3.1`. Still requires to be published on master. I didn't use master because has some breaking changes, so we would need to release micrate `v0.4.0`

Please see: https://github.com/amberframework/micrate/releases/tag/v0.3.1

**Comment by faustinoaq** [comment]
Fixed by #33

**Comment by eliasjpr** [comment]
Is this done? Would like to close this ticket and draft a new release
